Calculator.php

```diff
@@ -151,24 +152,65 @@ function show_history($history)
         info('You have no history');
     } else {
         $historyGroups = array_group($history, 'date');
+        $historyCommands = ['full', 'help'];
 
-        foreach ($historyGroups as $date => $historyItems) {
-            info ("{$date}:");
+        foreach ($historyGroups as $mainDate => $dateGroup) {
+            $historyCommands[] = $mainDate;
+        }
 
-            foreach ($historyItems as $historyItem) {
-                $isBasicMathOperation = in_array($historyItem['sign'], BASIC_COMMANDS);
+        $showDateHistory = readline('Enter date of history (day - month - year), "full" or "Help"  : ');
 
-                $prefix = ($isBasicMathOperation) ? '   ' : '(!)';
+        $int = filter_var($showDateHistory, FILTER_SANITIZE_NUMBER_INT);
+        $isDate = ($showDateHistory === $int);
 
-                $historyFunction = "{$prefix} {$historyItem['first_operand']} {$historyItem['sign']} {$historyItem['second_operand']} = {$historyItem['result']}";
+        if (!$isDate && !in_array($showDateHistory, $historyCommands)) {
+            info('Dont write letters.');
 
-                info($historyFunction, 1);
-            }
+            return;
+        }
+
+        if ($showDateHistory == 'help') {
+            show_info_block();
+        }
+
+        info('', 1);
+
+        if(!in_array($showDateHistory, $historyCommands)) {
+            info('You have no history in that day');
+
+            return;
+        }
+
+        if (array_key_exists($showDateHistory, $historyGroups)) {
+            info("{$showDateHistory}: ");
+
+            addprefix($historyGroups[$showDateHistory]);
 
             info('=====================');
+
+        } elseif ($showDateHistory == 'full') {
+            foreach ($historyGroups as $date => $historyItems) {
+                info("{$date}:");
+
+                addprefix($historyItems);
+
+                info('=====================');
```

Comment:
you can use write_symbol_line method here, also pelase move it to the show history items method